File: behavioral/observer/push.java
public interface Observer {
    void update(Video video);
}

public class SubscriberA implements Observer{
    public void update(Video video) {
        System.out.println("SubscriberA Notofications ");
    }
}

public class SubscriberB implements Observer {
    public void update(Video video) {
        System.out.println("SubscriberB Notofications ");
    }
}

public class SubscriberC implements Observer {
    public void update(Video video) {
        System.out.println("SubscriberC Notofications ");
    }
}

public interface Subject {
    void attach(Observer observer);
    void detach(Observer observer);
    void notifyObserver();
}

public class Youtube implements Subject {
    private List<Observer> observers;
    private Video video;

    public Youtube(List<Observer> observers) {
        this.observers = observers;
    }

    @Override
    public void attach(Observer observer) {
        observers.add(observer);
    }

    @Override
    public void detach(Observer observer) {
        observers.remove(observer);
    }

    @Override
    public void notifyObserver() {
        for (Observer observer: observers) {
            observer.update(video);
        }
    }

    public void newVideo(Video video) {
        this.video = video;
        notifyObserver();
    }
}

File: behavioral/observer/other_examples/push.py
from abc import ABCMeta, abstractmethod
from typing import List

class Video:
    pass

class Observer(metaclass=ABCMeta):
    @abstractmethod
    def update(self, video: Video):
        pass

class SubscriberA(Observer):
    def update(self, video: Video):
        print('SubscriberA Notifications')

class SubscriberB(Observer):
    def update(self, video: Video):
        print('SubscriberB Notifications')

class SubscriberC(Observer):
    def update(self, video: Video):
        print('SubscriberC Notifications')

class Subject(metaclass=ABCMeta):
    @abstractmethod
    def attach(self, observer: Observer):
        pass

    @abstractmethod
    def detach(self, observer: Observer):
        pass

    @abstractmethod
    def notify(self):
        pass

class Youtube(Subject):
    def __init__(self, observers: List[Observer]):
        self.observers = observers
        self.video = None

    def attach(self, observer: Observer):
        self.observers.append(observer)

    def detach(self, observer: Observer):
        self.observers.remove(observer)

    def notify(self):
        for observer in self.observers:
            observer.update(self.video)

    def new_video(self, video: Video):
        self.video = video
        self.notify()
